### ISSUE TITLE
Some linters require a file extension to run properly

### ISSUE THREAD

**Issue by cixtor** [SublimeLinter/SublimeLinter#912]
I was editing some PHP files the other day and then noticed that the _"SublimeLinter-phpcs"_ was not reporting any errors nor warnings over a file that clearly contained multiple coding style errors suggested by PSR2. After some time debugging the code I found the reason:

* SublimeText is completely off _(Cmd + Q on macOS)_
* Open SublimeText and then load a PHP file with PSR2 errors
* `SublimeLinter: detected syntax: plain text` is printed to the console
* Use the command prompt to change the syntax to PHP
* `SublimeLinter: detected syntax: html` is printed to the console
* Use the command prompt to change the syntax to PHP _(again)_
* `SublimeLinter: detected syntax: plain text` is printed to the console

This really doesn't matter much considering that the [PHP CodeSniffer Linter](https://github.com/SublimeLinter/SublimeLinter-phpcs/blob/a99189a/linter.py#L19) supports PHP, HTML and HTML5 syntax, however, after some thinking I realized what the problem was. PHP CodeSniffer — itself — doesn't runs on non-PHP files:

### Sample File

```
<?php

if
	(TRUE)
{
	echo "hello";
}

else

{echo "world";}
```

### PHP CodeSniffer Against A File With PHP Extension

```
$ phpcs --report=checkstyle --standard=PSR2 untitled.php 
  <?xml version="1.0" encoding="UTF-8"?>
  <checkstyle version="3.2.2">
  <file name="/Users/cixtor/Desktop/untitled.php">
    <error line="3" column="1" severity="error" message="[…]"/>
    <error line="4" column="1" severity="error" message="[…]"/>
    <error line="4" column="6" severity="error" message="[…]"/>
    <error line="4" column="10" severity="error" message="[…]"/>
    <error line="6" column="1" severity="error" message="[…]"/>
    <error line="7" column="1" severity="error" message="[…]"/>
    <error line="9" column="1" severity="error" message="[…]"/>
    <error line="11" column="1" severity="error" message="[…]"/>
    <error line="11" column="15" severity="error" message="[…]"/>
    <error line="11" column="15" severity="error" message="[…]"/>
  </file>
  </checkstyle>
```

### PHP CodeSniffer Against A File Without PHP Extension

```
$ phpcs --report=checkstyle --standard=PSR2 untitled
  <?xml version="1.0" encoding="UTF-8"?>
  <checkstyle version="3.2.2">
  </checkstyle>
```

When a linter runs, SublimeLinter creates a temporary file with the content of the file that is currently being edited. On macOS the files are usually stored under `$TMPDIR` and the names follow this pattern `tmp[a-z]{6}`. When the SublimeLinter-phpcs passes the command through the `cmd` variable, it ends up [here](https://github.com/SublimeLinter/SublimeLinter/blob/dde0412/lint/util.py#L299-L306) with the following:

```
# cmd = [
#     '/usr/local/bin/phpcs',
#     '--report=checkstyle',
#     '--standard=PSR2',
#     '/var/folders/gy/scj0933963ggbfdxcn5c91vw0000gn/T/tmpbilqso'
# ]
out = popen(cmd, stdout=stdout, stderr=stderr,
    output_stream=output_stream, env=env, cwd=cwd)
```

Because of this, SublimeLinter is unable to obtain the correct output and so SublimeLinter-phpcs cannot tag the errors.

### Solution

When SublimeLinter detects the file syntax, this information must be passed down to `util.communicate()`. Once the content of the original file is copied to the temporary file, rename the later to include the file extension, then SublimeLinter can execute the command via `process.Popen`.
